### PR TITLE
Feature/pantry class

### LINKED ISSUE #52: Create 'Pantry' class with accompanying test file
- [x] Determine if pantry has enough ingredients to cook a given meal
- [x] Determine the amount of missing ingredients still needed to cook a given meal, based on what’s in the user’s pantry.
- [x] As a user, I should be able to view what ingredients exist inside of my pantry.
- [x] As a user, I should be able to check my list of recipes to cook and see if my pantry has enough ingredients to cook a meal.
- [x] As a user, I should be told what ingredients are still needed if I don’t have enough ingredients in my pantry to cook the recipe.
- [x] As a user, I should not be able to cook a recipe if I don’t have the ingredients required.
- [x] As a user, when I cook a meal, those ingredients should be removed from my pantry.
- [x] As a user, I should be able to add more ingredients to my pantry.

### PR DESCRIPTION
**Pull request by philalewis**
What (if any) features are you implementing?
- Add pantry class and test file

What (if anything) did you refactor?
- Added one ingredient to the ingredients test data file
- Removed a bunch of ingredients from the user test data file

Were there any issues that arose?
- Nope

Is there anything that you need from your teammate?
- Please review and merge

Any other comments, questions, or concerns?
- We may have to refactor the User class to incorporate the Pantry class

### CODE REVIEW

File: src/classes/Pantry.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import Ingredient from './Ingredient';
+
+class Pantry {
+  constructor(pantry, data) {
+    this.ingredients = this.buildPantry(pantry, data);
+    this.ingredientsData = data;
+  }
+
+  buildPantry (pantry, data) {
```
Comment (mr-ryan12) [inline src/classes/Pantry.js:9]: Good use of the iterator method to build the pantry!

File: src/classes/Pantry.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      const thisIngredient = this.ingredients.find(ing => {
+        return ing.id === ingredient.id
+      })
+      !ids.includes(ingredient.id) || thisIngredient.quantity.amount < ingredient.quantity.amount ? canMake = false : null;
+    })
+
+    return canMake;
+  }
+
+  determineMissingIngredients(recipe) {
```
Comment (mr-ryan12) [inline src/classes/Pantry.js:32]: Nice use of ternary operators here!

File: src/data/users-test-data.js [hunk 3]
```diff
@@ -3,145 +3,25 @@ const usersData = [
   "name": "Saige O'Kon",
   "id": 1,
   "pantry": [
-    {
```
Comment (oacortez) [inline src/data/users-test-data.js:6]: Thank you for cleaning this users test data down. We definitely did not need all of those ingredients elements in user pantry.

File: src/classes/Pantry.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+          },
+          this.ingredientsData
+          )) : null;
+          return acc;
+        }, [])
+
+    return result;
+  }
+
+  addIngredient(ingredient) {
```
Comment (mr-ryan12) [inline src/classes/Pantry.js:50]: Thank you for getting this functionality to work. All tests passed!

File: test/Pantry-test.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import { expect } from 'chai';
```
Comment (mr-ryan12) [inline test/Pantry-test.js:1]: Good test coverage. We can revisit adding more sad path testing in the future if necessary.

File: test/User-test.js [hunk 6]
```diff
@@ -56,8 +56,8 @@ describe('User', () => {
 
   it('should have a pantry', () => {
     expect(user.pantry).to.be.an('array');
-    expect(user.pantry[0].amount).to.equal(4);
```
Comment (mr-ryan12) [inline test/User-test.js:59]: Good catch!

**Comment by mr-ryan12** [comment]
Thank you for your implementations! Next steps include getting our data from the localhost and moving our DOM manipulation into the domUpdates.js file.

**Pull request merged**